ui/src/shared/components/GetVariables.tsx

```diff
@@ -0,0 +1,57 @@
+// Libraries
+import React, {useEffect, FunctionComponent} from 'react'
+import {connect} from 'react-redux'
+
+// Components
+import {SpinnerContainer, TechnoSpinner} from '@influxdata/clockface'
+
+// Actions
+import {getVariables} from 'src/variables/actions'
+
+// Types
+import {RemoteDataState} from 'src/types'
+import {AppState} from 'src/types/v2'
+
+interface StateProps {
+  variablesStatus: RemoteDataState
+}
+
+interface DispatchProps {
+  onGetVariables: typeof getVariables
+}
+
+type Props = StateProps & DispatchProps
+
+const GetVariables: FunctionComponent<Props> = ({
+  variablesStatus,
+  onGetVariables,
+  children,
+}) => {
+  useEffect(() => {
+    if (variablesStatus === RemoteDataState.NotStarted) {
+      onGetVariables()
```

Comment:
reluctant to use `useEffect` for data fetching until the suspense api has been released. especially since a component that fetches arbitrary resources and calls the thunks that sets them on redux state already exists.  See `GetResources`

Reply:
It does seem likely that any data fetching in hooks we start to do now will be at odds with suspense patterns. I think we're in the same spot with how we fetch data in class components, but I can see value in sticking to just one style, even if it's also incompatible.  So for this PR, I'll replace this hook usage with `GetResources`.

The compatibility with redux with respect to suspense is also pretty unclear—I was reading [this issue](https://github.com/reduxjs/react-redux/issues/1177) the other day, which lays out some of the challenges that the react-redux lib is facing. It's an interesting read.

Reply:
Yeah, tbh, the aspect of "Redux and compat with Suspense" is _very_ unclear atm.  

Dan Abramov had some [thoughts on hypothetical gradiations of "compatibility"](https://github.com/reduxjs/react-redux/issues/1164#issuecomment-457196712).  Not sure where we're going to end up on that list, and I don't think we can even really figure that out until Suspense itself is truly ready for release.

I'd suggest writing code in whatever way works best for you now, and deal with Suspense once it's actually out.  Current patterns won't suddenly stop working - it's just that there may be a new alternative that _could_ simplify things.

Reply:
Gotcha. I think we're also trying to grapple with the multitude of different, yet functionally equivalent patterns for fetching data with Redux as it stands today. Trying to orient ourselves towards future compatibility with Suspense does seem like a bit much to take on at once.

I appreciate you chiming in, and am looking forward to where Redux ends up!